### PR TITLE
[feat]: [CDC-16674]: manifest fetch from artifactory before execution

### PR DESCRIPTION
**Pull request by prabu-kr**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/30558)

### CODE REVIEW

**Review by poojaSinghal — CHANGES_REQUESTED** [review]
> Reviewed 24 of 24 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30558)* status: 1 change requests. 0 of 1 approvals obtained. 

___
*[260-delegate/src/main/java/io/harness/perpetualtask/manifest/ArtifactoryHelmRepositoryService.java, line 47 at r1](https://reviewable.io/reviews/harness/harness-core/30558#-MvSCq963xOwS7zQmVgY:-MvSCq963xOwS7zQmVgZ:b-884we2) ([raw file](https://github.com/harness/harness-core/blob/99f096e64c29fd5cfb5dd5379617adb041acfa42/260-delegate/src/main/java/io/harness/perpetualtask/manifest/ArtifactoryHelmRepositoryService.java#L47)):*
> ```Java
> 
>     String repoName = artifactoryHelmTaskHelper.getArtifactoryRepoNameFromHelmConfig(helmRepoConfig);
>     artifactoryService.checkIfValidHelmRepository(request, repoName);
> ```

We should remove this 
___
*[260-delegate/src/main/java/software/wings/delegatetasks/helm/HelmValuesFetchTask.java, line 79 at r1](https://reviewable.io/reviews/harness/harness-core/30558#-MvSHDzZ2zCTvIMuke-1:-MvSHDzZ2zCTvIMuke-2:b-3ncumi) ([raw file](https://github.com/harness/harness-core/blob/99f096e64c29fd5cfb5dd5379617adb041acfa42/260-delegate/src/main/java/software/wings/delegatetasks/helm/HelmValuesFetchTask.java#L79)):*
> ```Java
>       HelmChartConfigParams helmChartConfigParams = taskParams.getHelmChartConfigTaskParams();
> 
>       if (helmChartConfigParams.isBypassHelmFetch()
> ```

Remove if not needed now. 
___
*[400-rest/src/main/java/software/wings/beans/appmanifest/ApplicationManifestSummary.java, line 25 at r1](https://reviewable.io/reviews/harness/harness-core/30558#-MvSHQEP-OZ6iujV4pEB:-MvSHQEP-OZ6iujV4pEC:b-hfpv2h) ([raw file](https://github.com/harness/harness-core/blob/99f096e64c29fd5cfb5dd5379617adb041acfa42/400-rest/src/main/java/software/wings/beans/appmanifest/ApplicationManifestSummary.java#L25)):*
> ```Java
> public class ApplicationManifestSummary {
>   private String appManifestId;
>   private String appManifestName;
> ```

remove 



_Code quote:_
```Java
appManifestName;
```
___
*[400-rest/src/main/java/software/wings/service/impl/WorkflowExecutionServiceImpl.java, line 2493 at r1](https://reviewable.io/reviews/harness/harness-core/30558#-MvSKJvp8td7etPdDxE4:-MvSKJvp8td7etPdDxE5:b17v4ih) ([raw file](https://github.com/harness/harness-core/blob/99f096e64c29fd5cfb5dd5379617adb041acfa42/400-rest/src/main/java/software/wings/service/impl/WorkflowExecutionServiceImpl.java#L2493)):*
> ```Java
>             helmChart.getAppId(), helmChart.getApplicationManifestId())));
>     versionInputHelmCharts.forEach(helmChart -> {
>       Map<String, String> properties = applicationManifestService.fetchAppManifestProperties(
> ```

Refactor 



_Code quote:_
```Java
 helmChart.setMetadata(properties
```
___
*[400-rest/src/main/java/software/wings/service/impl/WorkflowExecutionServiceImpl.java, line 2505 at r1](https://reviewable.io/reviews/harness/harness-core/30558#-MvSKVJ61a8_MkeTL5er:-MvSKVJ7Ca5lce1z3c9d:b-4s60uz) ([raw file](https://github.com/harness/harness-core/blob/99f096e64c29fd5cfb5dd5379617adb041acfa42/400-rest/src/main/java/software/wings/service/impl/WorkflowExecutionServiceImpl.java#L2505)):*
> ```Java
>     if (helmCharts == null || helmChartIds.size() + versionInputHelmCharts.size() != helmCharts.size()) {
>       log.error("helmChartIds from executionArgs contains invalid helmCharts");
>       throw new InvalidRequestException("Invalid helm chart");
> ```

what exactly was invalid should be added to error message



_Code quote:_
```Java
ion("Invalid helm chart");
```
___
*[400-rest/src/main/java/software/wings/service/impl/WorkflowExecutionServiceImpl.java, line 3075 at r1](https://reviewable.io/reviews/harness/harness-core/30558#-MvSLfDn3qDJhKcTf4bb:-MvSLfDn3qDJhKcTf4bc:b-al6u1d) ([raw file](https://github.com/harness/harness-core/blob/99f096e64c29fd5cfb5dd5379617adb041acfa42/400-rest/src/main/java/software/wings/service/impl/WorkflowExecutionServiceImpl.java#L3075)):*
> ```Java
>   }
> 
>   private void setManifestsFromManifestVariables(String appId, ExecutionArgs executionArgs, String accountId) {
> ```

is this just copying data from manifest VAriable to Helm Chart?? 
then we can copy it every time. 
___
*[400-rest/src/main/java/software/wings/service/impl/applicationmanifest/HelmChartServiceImpl.java, line 68 at r1](https://reviewable.io/reviews/harness/harness-core/30558#-MvSJknQ1a8QXVkNXDF5:-MvSJknQ1a8QXVkNXDF6:b-ew0g0a) ([raw file](https://github.com/harness/harness-core/blob/99f096e64c29fd5cfb5dd5379617adb041acfa42/400-rest/src/main/java/software/wings/service/impl/applicationmanifest/HelmChartServiceImpl.java#L68)):*
> ```Java
>     HelmChart existingHelmChart = getManifestByVersionNumber(
>         helmChart.getAccountId(), helmChart.getApplicationManifestId(), helmChart.getVersion());
>     if (existingHelmChart != null) {
> ```

move this get call outside of create 
___
*[400-rest/src/main/java/software/wings/sm/WorkflowStandardParams.java, line 243 at r1](https://reviewable.io/reviews/harness/harness-core/30558#-MvSNs48CYCVy5CG5oIk:-MvSNs4DCPF5SN0s7g3m:bidmhqq) ([raw file](https://github.com/harness/harness-core/blob/99f096e64c29fd5cfb5dd5379617adb041acfa42/400-rest/src/main/java/software/wings/sm/WorkflowStandardParams.java#L243)):*
> ```Java
>     }
> 
>     if (featureFlagService.isEnabled(FeatureName.BYPASS_HELM_FETCH, getApp().getAccountId())) {
> ```

only have helm chart Ids in WorkflowStandard Params

**Review by inderschera — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30558)* status: 15 file(s) to review. 1 change requests. 2. unresolved discussions. 0 of 1 approvals obtained. 

___
*[260-delegate/src/main/java/io/harness/perpetualtask/manifest/ArtifactoryHelmRepositoryService.java, line 47 at r1](https://reviewable.io/reviews/harness/harness-core/30558#-MvSCq963xOwS7zQmVgY:-Mvct9jd7XWfwELAzSXC:bs7wumx) ([raw file](https://github.com/harness/harness-core/blob/99f096e64c29fd5cfb5dd5379617adb041acfa42/260-delegate/src/main/java/io/harness/perpetualtask/manifest/ArtifactoryHelmRepositoryService.java#L47)):*
<details><summary><i>Previously, poojaSinghal (Pooja Singhal) wrote…</i></summary><blockquote>

We should remove this 
</blockquote></details>

Done. 
___
*[400-rest/src/main/java/software/wings/beans/appmanifest/ApplicationManifestSummary.java, line 25 at r1](https://reviewable.io/reviews/harness/harness-core/30558#-MvSHQEP-OZ6iujV4pEB:-MvctF1t0_YaocwL7jzA:b-wmas3v) ([raw file](https://github.com/harness/harness-core/blob/99f096e64c29fd5cfb5dd5379617adb041acfa42/400-rest/src/main/java/software/wings/beans/appmanifest/ApplicationManifestSummary.java#L25)):*
<details><summary><i>Previously, poojaSinghal (Pooja Singhal) wrote…</i></summary><blockquote>

remove 
</blockquote></details>

UI needs it. Keeping it

File: 400-rest/src/main/java/software/wings/service/impl/applicationmanifest/HelmChartServiceImpl.java [hunk 1]
```diff
@@ -217,7 +233,14 @@ public HelmChart fetchByChartVersion(
   }
 
   @Override
-  public List<HelmChart> fetchChartsFromRepo(String accountId, String appId, String serviceId, String appManifestId) {
+  public List<HelmChart> fetchChartsFromRepo(
+      String accountId, String appId, String serviceId, String appManifestId, boolean collectUsingDelegate) {
+    if (!collectUsingDelegate) {
+      ApplicationManifest applicationManifest = applicationManifestService.getById(appId, appManifestId);
+      if (Boolean.TRUE.equals(applicationManifest.getEnableCollection())) {
```
Comment (inderschera) [inline 400-rest/src/main/java/software/wings/service/impl/applicationmanifest/HelmChartServiceImpl.java:240]: change it to != false

**Review by prabu-kr — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30558)* status: 29 file(s) to review. 1 change requests. 2. unresolved discussions. 0 of 1 approvals obtained. 

___
*[260-delegate/src/main/java/io/harness/perpetualtask/manifest/ArtifactoryHelmRepositoryService.java, line 48 at r12](https://reviewable.io/reviews/harness/harness-core/30558#-MwGsWyn3BXKKFo5AwRE:-MwGsWyo5k29sll_U0Cn:b68qxgt) ([raw file](https://github.com/harness/harness-core/blob/3bcd6fa488658a032698a6ef3688001812075c27/260-delegate/src/main/java/io/harness/perpetualtask/manifest/ArtifactoryHelmRepositoryService.java#L48)):*
> ```Java
> 
>     List<HelmChart> helmCharts = artifactoryService.getHelmCharts(
>         request, repoName, helmChartCollectionParams.getHelmChartConfigParams().getChartName(), 50);
> ```

Constant variable for the size (50)?
___
*[400-rest/src/main/java/software/wings/helpers/ext/helm/request/HelmChartCollectionParams.java, line 47 at r12](https://reviewable.io/reviews/harness/harness-core/30558#-MwGtwak5SmEM0kvFsOk:-MwGtwak5SmEM0kvFsOl:b-67scdv) ([raw file](https://github.com/harness/harness-core/blob/3bcd6fa488658a032698a6ef3688001812075c27/400-rest/src/main/java/software/wings/helpers/ext/helm/request/HelmChartCollectionParams.java#L47)):*
> ```Java
>     List<ExecutionCapability> executionCapabilities =
>         helmChartConfigParams.fetchRequiredExecutionCapabilities(maskingEvaluator);
>     if (helmChartConfigParams.isBypassHelmFetch()
> ```

Extract out the artifactory checking logic in a common place? Maybe artifactoryHelmHelper?

**Review by poojaSinghal — DISMISSED** [review]
:lgtm:

> Reviewed 1 of 1 files at r3, 2 of 12 files at r4, 1 of 25 files at r10, 1 of 25 files at r11, 25 of 25 files at r12, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30558#-:-MwHC2w-Dw-ggiO3Zcfm:bnfp4nl)* status: :shipit: complete! ✓

**Review by inderschera — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30558)* status: 4 file(s) to review. 0 of 1 approvals obtained. 

___
*[260-delegate/src/main/java/io/harness/perpetualtask/manifest/ArtifactoryHelmRepositoryService.java, line 48 at r12](https://reviewable.io/reviews/harness/harness-core/30558#-MwGsWyn3BXKKFo5AwRE:-MwNDMYi5M3FDYiaClUY:bb6un1j) ([raw file](https://github.com/harness/harness-core/blob/3bcd6fa488658a032698a6ef3688001812075c27/260-delegate/src/main/java/io/harness/perpetualtask/manifest/ArtifactoryHelmRepositoryService.java#L48)):*
<details><summary><i>Previously, prabu-kr (Prabu Rajendran) wrote…</i></summary><blockquote>

Constant variable for the size (50)?
</blockquote></details>

Done
___
*[400-rest/src/main/java/software/wings/helpers/ext/helm/request/HelmChartCollectionParams.java, line 47 at r12](https://reviewable.io/reviews/harness/harness-core/30558#-MwGtwak5SmEM0kvFsOk:-MwNDILxDVnEoAVVp3_M:bb6un1j) ([raw file](https://github.com/harness/harness-core/blob/3bcd6fa488658a032698a6ef3688001812075c27/400-rest/src/main/java/software/wings/helpers/ext/helm/request/HelmChartCollectionParams.java#L47)):*
<details><summary><i>Previously, prabu-kr (Prabu Rajendran) wrote…</i></summary><blockquote>

Extract out the artifactory checking logic in a common place? Maybe artifactoryHelmHelper?
</blockquote></details>

Done

**Comment by inderschera** [comment]
trigger messagecheck

**Review by poojaSinghal — COMMENTED** [review]
> Reviewed 4 of 4 files at r13, 2 of 2 files at r14, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30558)* status: 0 of 1 approvals obtained.

**Pull request merged**